R/app-server.R

```diff
@@ -9,6 +9,18 @@
 #' @param session Shiny session
 #' @export
 app_server <- function(input, output, session) {
+  ## Render vignette markdown
+  output$markdown <- renderUI({
+    HTML(markdown::markdownToHTML(
+      knitr::knit(input = glue::glue(
+        tools::file_path_sans_ext(config::get("path_to_markdown")), ".Rmd"
+        ),
+        output = glue::glue(
+          tools::file_path_sans_ext(config::get("path_to_markdown")), ".md"
+        ),
+                  quiet = TRUE),
+      stylesheet = ("inst/app/www/custom.css")))
+  })
```

Comment:
I thought I left this comment a while ago, but it seems to not be here.

One of the issues mentioned was that the markdown takes time to render and, since it doesn't rely on anything needed from the user, could be done in `app_ui()` to decrease user wait time. To do so, this would be removed from the server function and put into a separate function (perhaps in app-ui.R) without the renderUI call: https://github.com/kelshmo/dccvalidator/blob/0f947f5b8b21328c786d6e1239574fd5629a644b/R/app-server.R#L12-L23

New function example:

```
get_markdown <- function() {
    HTML(markdown::markdownToHTML(
        knitr::knit(input = glue::glue(
            tools::file_path_sans_ext(config::get("path_to_markdown")), ".Rmd"
        ),
        output = glue::glue(
            tools::file_path_sans_ext(config::get("path_to_markdown")), ".md"
        ),
        quiet = TRUE)
}
```

Then the `app_ui()` would call the new function to generate the markdown and get the html.
```
tabItem(
          tabName = "vignette",
          get_markdown()
        ),
```

I believe this should work and would put the rendering burden on the initial spin-up of the app, as opposed to re-rendering for every user.

Reply:
This works locally, thanks @Aryllen ! @karawoo what do you think about implementing this change?

Reply:
Yeah that sounds great!